### DISCUSSION TITLE
Adding Company Portal Top-Level Navbar

### DISCUSSION THREAD

**Discussion by societymartingale** [facebook/docusaurus#8913] [Q&A] [answered]
Was hoping to get some advice on this issue, which I'm stuck on.

I am hosting a Docusaurus site on an internal company portal. I would like to include that portal's navbar at the top and the entire Docusaurus site (including the Docusaurus navbar) below that, so there would be 2 navbars stacked. So the basic layout would be:

```
[company portal navbar]
[Docusaurus navbar]
[Docusaurus site including landing page and docs]
```

The portal navbar is being pulled in via a server-side include in a React component:
```js
<div
  dangerouslySetInnerHTML={{
    __html: '<!--# include virtual="/devportal/header" -->',
  }}
/>
```

The other challenge is that the portal navbar uses a different css library, so I'm isolating it in the shadow dom, so it doesn't conflict with Infima css.

Has anyone done something similar to this or have any ideas on how to do this? Also, for swizzling in this component, what would be a good Docusaurus component to swizzle? I was initially thinking the Layout component.

**Comment by slorber** [comment, marked as answer]
> The portal navbar is being pulled in via a server-side include in a React component:

I don't know how this works but it's quite proprietary solution, we can't really help you here.

If your enterprise navbar is not using React and scoped CSS, I'd suggest starting to maintain another version that integrates well with React as a standalone component that you could use in Docusaurus. Trying to reuse what you already have is not always the best option.

Also, is it important that the corporate navbar is pixel-perfect the same as on other pages of your site? It could be good enough if it almost looks the same. For example, Figma uses Docusaurus for widgets/plugins and has a navbar that is not exactly the same compared to other sections, and it doesn't look too bad: https://www.figma.com/widget-docs/

> The other challenge is that the portal navbar uses a different css library, so I'm isolating it in the shadow dom, so it doesn't conflict with Infima CSS.

It's not clear to me the challenges you are facing. Using shadow dom can indeed be a solution to isolate the CSS.


> Has anyone done something similar to this or have any ideas on how to do this? Also, for swizzling in this component, what would be a good Docusaurus component to swizzle? I was initially thinking the Layout component.

I suggest to swizzle the Navbar component, and making it so that it becomes a combination of your enterprise + Docusaurus navbar

```tsx
import React from 'react';
import NavbarLayout from '@theme/Navbar/Layout';
import NavbarContent from '@theme/Navbar/Content';

export default function Navbar(): JSX.Element {
  return (
  <>
    <nav style={{border: "solid",padding: 10}}>Enterprise navbar</nav>
    <NavbarLayout>
      <NavbarContent />
    </NavbarLayout>
  </>
  );
}

```

<img width="1915" alt="CleanShot 2023-04-21 at 11 46 38@2x" src="https://user-images.githubusercontent.com/749374/233604863-b7b38231-51ca-456e-b8ac-4950d2d12efd.png">

But it really depends on many UI factors, for example if you want the enterprise navbar to be "sticky" like the Docusaurus navbar etc... That's UI/layout design, there's no one size fit all solution and you'll have to figure out the details if your integration is complex.

**Reply by societymartingale** [reply]
Thanks @slorber for the detailed response and the code sample. Swizzling the Navbar component is working better. For some reason, the server-side include to display the company navbar isn't working when switching from the landing page to the docs pages (but reappears when refreshing the page). However, I think that a more general issue with React and server-side includes than with Docusaurus.
